app/controllers/external_activities_controller.rb

```diff
@@ -74,7 +74,16 @@ def show
       format.html {
         redirect_to(browse_external_activity_path(@external_activity))
       }
-      format.run_resource_html   { redirect_to(@external_activity.url) }
+      format.run_resource_html {
+        # automatically log in teachers
+        if current_visitor.portal_teacher
+           uri = URI.parse(@external_activity.url)
+           uri.query = URI.encode_www_form(URI.decode_www_form(uri.query || '') << ["domain", root_url] << ["domain_uid", current_visitor.id])
```

Comment:
I did it this way instead of setting the query in case the external activity url had a query parameter already.  However while testing this code I noticed it will throw an error if it thinks the query is malformed like 
?foo=bar&baz" (even through that is a valid query parameter)